aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -238,10 +248,7 @@ Tensor& normal_out_impl(Tensor& output, double mean, const Tensor& std, c10::opt
 
 template<template<typename> class normal_kernel, typename RNG>
 Tensor& normal_out_impl(Tensor& output, const Tensor& mean, const Tensor& std, c10::optional<Generator> gen) {
```

Comment:
Doesn't `mean` need a check here?

Reply:
not sure, i guess so. but i simply tried not to change the existing behavior

Reply:
not sure, i guess so. but i simply tried not to change the existing behavior

Reply:
I think those checks are done inside the other kernel calls: `mul` and `add`. So, maybe we don't really need them here...

Reply:
It doesn't look like `add_` or `mul_` fail there, probably because normal type promotion rules just promote the output to be complex:
```
>>> a = torch.ones(2)
>>> b = torch.ones(2, dtype=torch.complex64)
>>> torch.normal(a, b)  # this fails due to the check (complex std)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: normal expects standard deviation to be non-complex
>>> torch.normal(b, a) # but this works (complex mean)
tensor([-0.2736+0.7898j,  0.9972-0.1841j])
```

@anjali411 do you think it makes sense to add the complex check for `mean` here too?

Reply:
`mean` can be complex, but we should document the behaviour. In particular, here we are implementing
https://en.wikipedia.org/wiki/Complex_normal_distribution#Complex_normal_random_variable

Reply:
...if I'm not mistaken. We can do a plot of a number of samples to be sure.

Reply:
That sounds reasonable, although that should probably go as a separate PR (or be filed as a separate issue)

Reply:
@Lezcano is right; mean can be complex so this LGTM

Reply:
yep, let's do this separately

Reply:
so... i looked at the complex support yesterday. i think it should be in the separate pr as it would require refactoring the tests and adding support for testing more variants in `test_nn`. so i'd rather land this one as is and do the complex stuff later to avoid delaying this any further. so once i fix the ci issues, this should be good to go.

p.s. log_normal also has a mean parameter, which perhaps could also be complex (and maybe some other ops?). so it makes sense to do the complex work separately imo

Reply:
so that we don't forget https://github.com/pytorch/pytorch/issues/72134